### DISCUSSION TITLE
P2P sync

### DISCUSSION THREAD

**Discussion by sylverbee** [Mimetis/Dotmim.Sync#713] [Ideas]
I'm wondering how to adapt code to use peer-to-peer sync directly between users, instead of a server.
Any idea to start?

**Comment by Mimetis** [comment]
You can start from scratch I guess :)

**Reply by sylverbee** [reply]
My concern is how to exchange data between users without a centralized data base. I'd like to remove my SQL server. 
Any idea?

**Reply by Mimetis** [reply]
I know, but it will require to rework almost everything, especially the stored procedures / sql statements
I know that the old **MS Sync Framework 2.1** had this [option available](https://docs.microsoft.com/en-us/previous-versions/sql/synchronization/mt490617(v=msdn.10)), but it's quite old and deprecated today
Maybe it could be a good starting point to investigate.

Regarding **DMS**, it will require a lot of work, and the best solution would be to create a new project based on **DMS** and the refactoring feedback you get from the old **MS Sync Fx 2.1** peer to peer option.

**Reply by sylverbee** [reply]
Thx for the reply. I gonnoa investigate into OrbitDB or AvionDB

**Reply by sylverbee** [reply]
Have you already use rQlite?

**Reply by Mimetis** [reply]
Nop
